File: java/etl/src/main/java/Etl.java
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

class Etl {
    Map<String, Integer> transform(Map<Integer, List<String>> old) {
        return old.keySet()
                  .stream()
                  .flatMap(score -> old.get(score)
                                       .stream()
                                       .map(letter -> new KeyValuePair<>(letter.toLowerCase(), score)))
                  .collect(Collectors.toMap(kvPair -> kvPair.key, kvPair -> kvPair.value));
    }

    private static final class KeyValuePair<K, V> {
        final K key;
        final V value;

        KeyValuePair(K key, V value) {
            this.key = key;
            this.value = value;
        }
    }
}

File: javascript/etl/_iterations/1/etl.js
export function transform(old) {
    const scores = [].concat(...Object.keys(old).map(score => old[score].map(letter => [letter.toLowerCase(), score])));
    const newScores = {};
    scores.forEach(([letter, score]) => newScores[letter] = Number(score));
    return newScores;
}
